### PR TITLE
feat(package): add feature to execute manual deployments to destinations such as app stores

### PR DESCRIPTION
**Pull request by giocalitri**

### CODE REVIEW

File: packages/@ionic/cli/src/commands/package/deploy.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+export class BuildCommand extends Command {
+  async getMetadata(): Promise<CommandMetadata> {
+    const dashUrl = this.env.config.getDashUrl();
+
+    return {
+      name: 'build',
+      type: 'project',
+      summary: 'Deploys a package binary to an App Store using Appflow',
+      description: `
+This command deploys an existing package binary to an App Store using Ionic Appflow. While the deployment is running, it prints the remote log to the terminal.
```
Comment (lm3m) [inline packages/@ionic/cli/src/commands/package/deploy.ts:53]: maybe
```This command deploys a binary to a destination using Ionic Appflow. While the running, the remote log is printed to the terminal.```

File: packages/@ionic/cli/src/commands/package/deploy.ts [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+}
+
+export class BuildCommand extends Command {
+  async getMetadata(): Promise<CommandMetadata> {
+    const dashUrl = this.env.config.getDashUrl();
+
+    return {
+      name: 'build',
+      type: 'project',
+      summary: 'Deploys a package binary to an App Store using Appflow',
```
Comment (lm3m) [inline packages/@ionic/cli/src/commands/package/deploy.ts:51]: ```Deploys a binary to a destination, such as an app store using Appflow```

File: packages/@ionic/cli/src/commands/package/deploy.ts [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        {
+          id: 'dashboard',
+          url: dashUrl,
+        },
+      ],
+      exampleCommands: ['123456789 "My app store destination"'],
+      inputs: [
+        {
+          name: 'build-id',
+          summary: `The build id`,
```
Comment (lm3m) [inline packages/@ionic/cli/src/commands/package/deploy.ts:68]: ```The build id of the desired successful package build```

File: packages/@ionic/cli/src/commands/package/deploy.ts [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+          validators: [validators.required, validators.numeric],
+        },
+        {
+          name: 'destination',
+          summary:
+            'The destination to deploy the build artifact to the app store',
```
Comment (lm3m) [inline packages/@ionic/cli/src/commands/package/deploy.ts:74]: ```The destination where the build artifact should be deployed```

File: packages/@ionic/cli/src/commands/package/deploy.ts [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+      if (isSuperAgentError(e)) {
+        if (e.response.status === 401) {
+          this.env.log.error('Try logging out and back in again.');
+        }
+        const apiErrorMessage =
+          e.response.body.error && e.response.body.error.message
+            ? e.response.body.error.message
+            : 'Api Error';
+        throw new FatalException(
+          `Unable to create deployment build: ` + apiErrorMessage
```
Comment (lm3m) [inline packages/@ionic/cli/src/commands/package/deploy.ts:178]: is there a better way of saying this that might be actionable...

File: packages/@ionic/cli/src/commands/package/deploy.ts [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+  caller_id: number;
+  created: string;
+  state: string;
+  distribution_credential_name: string;
+  package_build: PackageBuild;
+  binary_deployment: BinaryDeployment;
+  distribution_trace: string;
+}
+
+export class BuildCommand extends Command {
```
Comment (dwieeb) [inline packages/@ionic/cli/src/commands/package/deploy.ts:44]: Small thing, but maybe rename to `DeployCommand`?

Comment (giocalitri) [reply packages/@ionic/cli/src/commands/package/deploy.ts:44]: sorry, copy/paste leftover

Comment (giocalitri) [reply packages/@ionic/cli/src/commands/package/deploy.ts]: pushed change

File: packages/@ionic/cli/src/commands/package/deploy.ts [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+export class DeployCommand extends Command {
+  async getMetadata(): Promise<CommandMetadata> {
+    const dashUrl = this.env.config.getDashUrl();
+
+    return {
+      name: 'build',
+      type: 'project',
+      summary: 'Deploys a binary to a destination, such as an app store using Appflow',
+      description: `
+This command deploys a binary to a destination using Ionic Appflow. While the running, the remote log is printed to the terminal.
```
Comment (dwieeb) [inline packages/@ionic/cli/src/commands/package/deploy.ts:53]:
```suggestion
This command deploys a binary to a destination using Ionic Appflow. While running, the remote log is printed to the terminal.
```

Comment (giocalitri) [reply packages/@ionic/cli/src/commands/package/deploy.ts]: pushed the change

**Pull request merged**